python/nav/portadmin/snmp/cisco.py

```diff
@@ -300,5 +315,88 @@ def get_dot1x_enabled_interfaces(self):
             for oid, state in self._bulkwalk(self.dot1xPortAuth)
         }
 
+    def get_poe_state_options(self) -> Sequence[PoeState]:
+        """Returns the available options for enabling/disabling PoE on this netbox"""
+        options_list = [
+            self.POE_AUTO,
+            self.POE_STATIC,
+            self.POE_LIMIT,
+            self.POE_DISABLE,
+        ]
+        return options_list
+
+    @translate_protocol_errors
+    def set_poe_state(self, interface: manage.Interface, state: PoeState):
+        """Set state for enabling/disabling PoE on this interface.
+        Available options should be retrieved using `get_poe_state_options`
+        """
+        unit_number, interface_number = self._get_poe_indexes_for_interface(interface)
+        oid_with_unit_number = self.POEENABLE + f".{unit_number}"
+        try:
+            self._set_netbox_value(
+                oid_with_unit_number, interface_number, 'i', state.state
+            )
+        except SnmpError as error:
+            _logger.error('Error setting poe state: %s', error)
+            raise
+        except ValueError as error:
+            _logger.error('%s is not a valid option for poe state', state)
+            raise
+
+    def _get_poe_indexes_for_interface(
+        self, interface: manage.Interface
+    ) -> Tuple[int, int]:
+        """Returns the unit number and interface number for the given interface"""
+        try:
+            poeport = manage.POEPort.objects.get(interface=interface)
+        except manage.POEPort.DoesNotExist:
+            raise POEIndexNotFoundError(
+                "This interface does not have PoE indexes defined"
+            )
+        unit_number = poeport.poegroup.index
+        interface_number = poeport.index
+        return unit_number, interface_number
+
+    def get_poe_states(
+        self, interfaces: Sequence[manage.Interface] = None
+    ) -> Dict[int, Optional[PoeState]]:
+        """Retrieves current PoE state for interfaces on this device.
+
+        :param interfaces: Optional sequence of interfaces to filter for, as fetching
+                           data for all interfaces may be a waste of time if only a
+                           single interface is needed. If this parameter is omitted,
+                           the default behavior is to filter on all Interface objects
+                           registered for this device.
+        :returns: A dict mapping interfaces to their discovered PoE state.
+                  The key matches the `ifindex` attribute for the related
+                  Interface object.
+                  The value will be None if the interface does not support PoE.
+        """
```

Comment:
I approved the base updates to the `ManagementHandler` API, but now I'm wondering if using ifindexes in response values was such a good idea.  The only other `ManagementHandler` methods that return a dict mapping for interfaces uses the interface names as the key, not their SNMP ifindex, which may be meaningless if the protocol handler isn't talking SNMP (like the Juniper handler, which speaks NETCONF-ish)

Reply:
So `Interface.ifname` should be used instead?

Reply:
I also have some apprehensions for how the interfaces argument is handled. 

Is this part really necessary:

```
If this (interfaces) parameter is omitted,
the default behavior is to filter on all Interface objects
registered for this device.
```
If the caller of the function wants to get poe state for all interfaces, then they can just put that in the `interfaces` argument directly.

Basically I am saying that maybe it shouldn't be optional at all